In [3]:
import pandas as pd

import warnings
warnings.simplefilter(action='ignore', category=FutureWarning)


df = pd.read_csv('all_ticker_misses.csv')

total_records = len(df)

dense_50 = len(df[df['miss_rate'] < 0.5])
dense_90 = len(df[df['miss_rate'] < 0.1])
dense_95 = len(df[df['miss_rate'] < 0.05])
dense_99 = len(df[df['miss_rate'] < 0.01])

print(f"- 50% dense: {dense_50} of {total_records} ({dense_50/total_records:.1%})")
print(f"- 90% dense: {dense_90} of {total_records} ({dense_90/total_records:.1%})")
print(f"- 95% dense: {dense_95} of {total_records} ({dense_95/total_records:.1%})")
print(f"- 99% dense: {dense_99} of {total_records} ({dense_99/total_records:.1%})")

- 50% dense: 2752 of 6564 (41.9%)
- 90% dense: 910 of 6564 (13.9%)
- 95% dense: 608 of 6564 (9.3%)
- 99% dense: 248 of 6564 (3.8%)


- 50% dense: 2752 of 6564 (41.9%)
- 90% dense: 910 of 6564 (13.9%)
- 95% dense: 608 of 6564 (9.3%)
- 99% dense: 248 of 6564 (3.8%)

This is dissapointing

In [ ]:
# write out the tickers of the 95% dense

dense_95_df = df[df['miss_rate'] < 0.05]
dense_95_df.to_csv('dense_95.csv', index=False)